fleet_adapter_mir/MiRCommandHandle.py

```diff
@@ -0,0 +1,986 @@
+            robot_state.battery_percent = api_response['battery_percentage']
+
+            robot_state.location = rmf_location
+            robot_state.path = self.rmf_robot_state_path_locations
+            robot_state.location.t.sec = now_sec
+            robot_state.location.t.nanosec = now_ns
+
+            if api_response['mission_text'].startswith('Charging'):  # Charging
+                robot_state.mode.mode = RobotMode.MODE_CHARGING
+                self.mir_state = MiRState.READY
+            elif api_response['state_id'] == MiRState.PAUSE:  # Paused/Pre-empted
+                self.pause()
+                robot_state.mode.mode = RobotMode.MODE_PAUSED
+                self.mir_state = MiRState.PAUSE
+            elif (api_response['state_id'] == MiRState.EXECUTING  # Moving
+                  and not api_response['mission_text'].startswith('Charging')):
+                self.resume()
+                robot_state.mode.mode = RobotMode.MODE_MOVING
+                self.mir_state = MiRState.EXECUTING
+            elif api_response['state_id'] == MiRState.READY:  # Idle/Moved
+                self.resume()
+                robot_state.mode.mode = RobotMode.MODE_IDLE
+                self.mir_state = MiRState.READY
+
+            if self.rmf_docking_requested:
+                if self.rmf_docking_executed:  # Docked
+                    if api_response['state_id'] == MiRState.READY:
+                        robot_state.mode.mode = RobotMode.MODE_IDLE
+                    else:  # Docking
+                        robot_state.mode.mode = RobotMode.MODE_DOCKING
+
+            # Update internal RobotState
+            self.robot_state = robot_state
+
+            self.last_robot_state_update = (
+                self.node.get_clock().now().nanoseconds / 1e9
+            )
+        except Exception as e:
+            self.node.get_logger().warn(f'Exception: {e}')
+
+    ##########################################################################
+    # INTERNAL UPDATE LOOP
+    ##########################################################################
+    def execute_updates(self, api_response=None):
+        if api_response is None:
+            api_response = self.mir_api.status_get()
+
+        self.update_internal_robot_state(api_response=api_response)
+        self.update_internal_location_trackers()
+        self.update_position(api_response=api_response)
+
+        self.state_update_timer.reset()
+
+
+    def lock_and_execute_updates(self):
+        with self._update_mutex:
+            self.execute_updates()
+
+
+    def get_map_name(self, graph_index):
+        self.node.get_logger().info(str(self.rmf_graph.get_waypoint(graph_index)))
+        return self.rmf_graph.get_waypoint(graph_index).map_name
+
+###############################################################################
+# HELPER FUNCTIONS AND CLASSES
+###############################################################################
+# class MiRRetryContext():
```

Comment:
Consider deleting this comment block, or uncommenting it if this class is potentially useful. If it's potentially useful but not something we want to commit ourselves to maintaining, then.. I guess it can remain as a comment block for now :shrug:

Reply:
https://github.com/osrf/fleet_adapter_mir/pull/14/commits/a5d7a33b1af5c996998be4e2bb10adf0ebcfe9b1